cmd/starbridge/model/chain.go

```diff
@@ -34,10 +42,32 @@ var (
 			AssetStellar_WUSDC.MapKey(): AssetEthereum_USDC,
 			AssetStellar_XLM.MapKey():   AssetEthereum_WXLM,
 		},
-		nextNonceFn: unsupportedNonceForChain, // TODO we haven't had the time to add the logic to go from Stellar to Ethereum yet
+		nextNonceFn:                  unsupportedNonceForChain, // TODO NS we haven't had the time to add the logic to go from Stellar to Ethereum yet
+		ValidateDestinationAddressFn: unsupportedValidateDestinationAddressFn,
 	}
 )
 
+func computeAllAssetMap(chain *Chain) map[string]*AssetInfo {
+	m := map[string]*AssetInfo{}
+	for _, v := range chain.AddressMappings {
+		m[v.MapKey()] = v
+	}
+
+	// native asset is included in the above list since it will be in the AddressMappings but add explicitly here too
+	m[chain.NativeAsset.MapKey()] = chain.NativeAsset
+
+	logger.Debugf("added %d items when creating AllAssetMap for chain=%s", len(m), chain.Name)
+	return m
+}
+
+func init() {
+	logger = supportlog.New()
+	logger.SetLevel(logrus.InfoLevel)
```

Comment:
💡 Similar request as earlier, can we make it a type where the logger can be provided as a dependency rather than having package level state?

Reply:
yes, will be fixing soon :)